cohere/client_async.py

```diff
@@ -105,6 +105,10 @@ async def _request(self, endpoint, json=None, method="POST", full_url=None, stre
             json_response = await response.json()
         except jsonlib.decoder.JSONDecodeError:  # CohereAPIError will capture status
             raise CohereAPIError.from_response(response, message=f"Failed to decode json body: {await response.text()}")
+        except aiohttp.ClientPayloadError as e:  # CohereAPIError will capture status
+            raise CohereAPIError.from_response(
+                response, message=f"An unexpected error occurred while receiving the response: {e}"
+            )
```

Comment:
I think it's slightly clearer. I didn't get the comment at first glance until I opened the code for `CohereAPIError`
```suggestion
	# `CohereAPIError.from_response()` will capture the http status code
        except jsonlib.decoder.JSONDecodeError:
            raise CohereAPIError.from_response(response, message=f"Failed to decode json body: {await response.text()}")
        except aiohttp.ClientPayloadError as e:
            raise CohereAPIError.from_response(
                response, message=f"An unexpected error occurred while receiving the response: {e}"
            )
```